tests/build/image_controller.go

```diff
@@ -0,0 +1,276 @@
+			if err != nil {
+				Expect(err.Error()).To(ContainSubstring("Reference does not exist"))
+			}
+
+			// Delete created webhook from GitHub
+			hooks, err := f.AsKubeAdmin.CommonController.Github.ListRepoWebhooks(imageControllerComponentGitSourceRepoName)
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, h := range hooks {
+				hookUrl := h.Config["url"].(string)
+				if strings.Contains(hookUrl, pacControllerHost) {
+					Expect(f.AsKubeAdmin.CommonController.Github.DeleteWebhook(imageControllerComponentGitSourceRepoName, h.GetID())).To(Succeed())
+					break
+				}
+			}
+		})
+
+		When("a component with image controller annotation delete-repo=true is created", func() {
+			BeforeAll(func() {
+				deleteRepo := true
+				// Create a component with Git Source URL and a specified git branch
+				_, err = f.AsKubeAdmin.HasController.CreateComponentWithPaCEnabled(applicationName, componentName, testNamespace, imageControllerComponentGitSourceURL, componentBaseBranchName, deleteRepo)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+			It("triggers a PipelineRun", func() {
+				timeout = time.Second * 600
+				interval = time.Second * 1
+				Eventually(func() bool {
+					pipelineRun, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(componentName, applicationName, testNamespace, "")
+					if err != nil {
+						GinkgoWriter.Println("PipelineRun has not been created yet")
+						return false
+					}
+					return pipelineRun.HasStarted()
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRun to start")
+			})
+			It("the PipelineRun should eventually finish successfully", func() {
+				timeout = time.Minute * 30
+				interval = time.Second * 10
+				Eventually(func() bool {
+					pipelineRun, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(componentName, applicationName, testNamespace, "")
+					Expect(err).ShouldNot(HaveOccurred())
+
+					for _, condition := range pipelineRun.Status.Conditions {
+						GinkgoWriter.Printf("PipelineRun %s Status.Conditions.Reason: %s\n", pipelineRun.Name, condition.Reason)
+
+						if !pipelineRun.IsDone() {
+							return false
+						}
+
+						if !pipelineRun.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
+							failMessage, err := tekton.GetFailedPipelineRunLogs(f.AsKubeAdmin.CommonController, pipelineRun)
+							if err != nil {
+								GinkgoWriter.Printf("failed to get logs for pipelinerun %s: %+v\n", pipelineRun.Name, err)
+							}
+							Fail(failMessage)
+						}
+					}
+					return true
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRun to finish")
+			})
+			It("image repo and robot account created successfully", func() {
+
+				component, err := f.AsKubeAdmin.HasController.GetHasComponent(componentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred(), "could not get component")
```

Comment:
please add more information to the error (componentName, testNamespace), and similarly to other assertions below

Reply:
Added more info in the assertion now.